airflow/gcp/operators/display_video.py

```diff
@@ -0,0 +1,301 @@
+        :ref:`howto/operator:GoogleDisplayVideo360DownloadReportOperator`
+
+    .. seealso::
+        Check also the official API docs:
+        `https://developers.google.com/bid-manager/v1/queries/getquery`
+
+    :param report_id: Report ID to retrieve.
+    :type report_id: str
+    :param bucket_name: The bucket to upload to.
+    :type bucket_name: str
+    :param report_name: The report name to set when uploading the local file.
+    :type report_name: str
+    :param chunk_size: File will be downloaded in chunks of this many bytes.
+    :type chunk_size: int
+    :param gzip: Option to compress local file or file data for upload
+    :type gzip: bool
+    :param api_version: The version of the api that will be requested for example 'v3'.
+    :type api_version: str
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate, if any. For this to work, the service accountmaking the
+        request must have  domain-wide delegation enabled.
+    :type delegate_to: str
+    """
+
+    template_fields = ("report_id", "bucket_name", "report_name")
+
+    @apply_defaults
+    def __init__(
+        self,
+        report_id: str,
+        bucket_name: str,
+        report_name: str,
+        gzip: bool = False,
+        chunk_size: int = 5 * 1024 * 1024,
+        api_version: str = "v1",
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: Optional[str] = None,
+        *args,
+        **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.report_id = report_id
+        self.chunk_size = chunk_size
+        self.gzip = gzip
+        self.bucket_name = self._set_bucket_name(bucket_name)
+        self.report_name = self._set_report_name(report_name)
+        self.api_version = api_version
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+
+    @staticmethod
+    def _set_report_name(name: str) -> str:
+        return name if name.endswith(".csv") else name + ".csv"
+
+    @staticmethod
+    def _set_bucket_name(name: str) -> str:
+        bucket = name if not name.startswith("gs://") else name[5:]
+        return bucket.strip("/")
+
+    def execute(self, context: Dict):
+        hook = GoogleDisplayVideo360Hook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            api_version=self.api_version,
+        )
+        gcs_hook = GoogleCloudStorageHook(
+            google_cloud_storage_conn_id=self.gcp_conn_id, delegate_to=self.delegate_to
+        )
+
+        self.log.info("Starting downloading report %s", self.report_id)
+        resource = hook.get_query(query_id=self.report_id)
+        file_url = resource["metadata"]["googleCloudStoragePathForLatestReport"]
+
+        # Download the report
+        with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+            with urllib.request.urlopen(file_url) as response:
```

Comment:
Are you sure this will not load the whole file into memory ? It likely does but I am not 100 % sure.  I guess these reports can be quite big so it might be worth doing streaming here.

Reply:
It seems to be a recommended way of doing things:
https://docs.python.org/3/howto/urllib2.html#fetching-urls

If I fully understand `urlopen` does not load the file into memory and `shutil.copyfileobj`:
>Copy the contents of the file-like object fsrc to the file-like object fdst. ... by default the data is read in chunks to avoid uncontrolled memory consumption.

https://docs.python.org/3/library/shutil.html#shutil.copyfileobj

Reply:
Good then :)